### DISCUSSION TITLE
is there any reason ctx.request.body is non-enumerable?

### DISCUSSION THREAD

**Discussion by aladdin-add** [koajs/koa#1759] [Q&A] [answered]
We are using Koa v2.13.4 and Koa-body v5.0.0, and when an error occurs, we want to record the request information. We used `JSON.stringify(ctx.request)` but the output did not include `ctx.request.body`. would like to understand why it is designed this way and if there is any recommended way to handle this.

**Comment by kleenkanteen** [comment]
dead repo

**Comment by jonathanong** [comment, marked as answer]
many of the fields in koa are getters/setters, which are not enumerable
